### PR TITLE
[Driver][SYCL][NFC] Update interface to access SYCL header additions

### PR DESCRIPTION
**Pull request by mdtoguchi**
Move the ability to add the SYCL headers into the SYCLInstallation class to better conform to existing usage for other toolchains.  Also improve the toolchain access for adding the headers using a virtual function.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/SYCL.cpp [hunk 1]
```diff
@@ -43,6 +43,30 @@ void SYCLInstallationDetector::getSYCLDeviceLibPath(
   DeviceLibPaths.emplace_back(D.SysRoot + "/lib");
 }
 
+void SYCLInstallationDetector::AddSYCLIncludeArgs(
+    const ArgList &DriverArgs, ArgStringList &CC1Args) const {
+  // Add the SYCL header search locations in the specified order.
+  //   ../include/sycl
+  //   ../include/sycl/stl_wrappers
+  //   ../include
```
Comment (bader) [inline clang/lib/Driver/ToolChains/SYCL.cpp:51]: @intel/llvm-reviewers-runtime, I suggest we remove the need for the driver to add 3 different locations for SYCL headers restricted by specific order. Considering that we expose only one header to the user API - `sycl/sycl.hpp`, the driver can include only one location and runtime library project can put everything into a single location or include from other locations. Right?

Comment (aelovikov-intel) [reply clang/lib/Driver/ToolChains/SYCL.cpp:51]: I don't know about `include/sycl` vs `include/`, but the `stl_wrappers` path has to be added in the driver.

Comment (bader) [reply clang/lib/Driver/ToolChains/SYCL.cpp:51]: Are there any conflicts between the files in `../include/sycl/stl_wrappers` and `../include`?

Comment (aelovikov-intel) [reply clang/lib/Driver/ToolChains/SYCL.cpp:51]: I believe maintainability benefits of having these headers in a separate dedicated folder outweigh the costs of having driver add one extra include path, if that's where you're going with your question.

Comment (bader) [reply clang/lib/Driver/ToolChains/SYCL.cpp:51]: What exactly is "maintainability benefits"? Note, these are directories of deployed compiler. We can still have separate directories in the sources code.

Comment (aelovikov-intel) [reply clang/lib/Driver/ToolChains/SYCL.cpp:51]: First, "smart" deploy is extra maintainability burden. Second, I want to be able to remove `stl_wrappers` from search path via manually modifying `clang -cc1`, merging them together with other includes would prevent that. I'm sure there are "third"/"fourth" if I were to think about it just a bit more.

**Comment by mdtoguchi** [comment]
@intel/dpcpp-clang-driver-reviewers, please take a look - thanks!

**Review by hchilama — APPROVED** [review]

**Comment by mdtoguchi** [comment]
@intel/llvm-gatekeepers, this should be ready for merge - thanks!

File: clang/lib/Driver/ToolChains/SYCL.cpp [hunk 2]
```diff
@@ -1824,29 +1848,9 @@ SYCLToolChain::GetCXXStdlibType(const ArgList &Args) const {
...
-  SmallString<128> STLWrappersPath(SYCLPath);
-  llvm::sys::path::append(STLWrappersPath, "stl_wrappers");
-  CC1Args.push_back("-internal-isystem");
-  CC1Args.push_back(DriverArgs.MakeArgString(SYCLPath));
-  CC1Args.push_back("-internal-isystem");
-  CC1Args.push_back(DriverArgs.MakeArgString(STLWrappersPath));
-  CC1Args.push_back("-internal-isystem");
-  CC1Args.push_back(DriverArgs.MakeArgString(IncludePath));
+void SYCLToolChain::AddSYCLIncludeArgs(const ArgList &DriverArgs,
+                                       ArgStringList &CC1Args) const {
```
Comment (srividya-sundaram) [inline clang/lib/Driver/ToolChains/SYCL.cpp:1852]: Should this have the 'override' specifier?

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/SYCL.cpp:1852]: AFAIK, the `override` is not needed for the function definition of the derived class, as the declaration states the `override`.

**Review by srividya-sundaram — APPROVED** [review]

**Comment by mdtoguchi** [comment]
Had to fix some conflicts, after update things are clean.  @intel/llvm-gatekeepers can this be merged?

**Pull request merged**